### ISSUE TITLE
adding or removing values from or into an array and then calling `get` returns  `_Map<String, dynamic>`

### ISSUE THREAD

**Issue by Nidal-Bakir** [parse-community/Parse-SDK-Flutter#834]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/main/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the latest version of [Parse Server](https://github.com/parse-community/parse-server/releases) and the [Parse Flutter SDK](https://github.com/parse-community/Parse-SDK-Flutter/releases). 

### Issue Description
Adding or removing values from or into an array using (`setAdd`, `setAddAll`, `setAddUnique`, `setAddAllUnique`, `setRemove`, `setRemoveAll`) in `ParseObject` and then calling `get` returns  `_Map<String, dynamic>`

related to: #696, #842 

### Steps to reproduce

1. create `ParseObject`
2. add a value to an array using `setAdd` function for example 
3. get the array using the array key with `get("key")`
4. the result of calling `get("key")` is `Map` which is not expected 

code snippet:
```dart 
final dietPlansObject = ParseObject("Diet_Plans");

dietPlansObject.setAdd("arrayKey", 1);

final array = dietPlansObject.get("arrayKey");

print(array.runtimeType); // _Map<String, dynamic>
``` 

### Actual Outcome
`_Map<String, dynamic>`

### Expected Outcome
An `Iterable` that contains all the added values and does not contain the removed ones

### Environment

Parse Flutter SDK
- SDK version: `3.1.15`
- Operating system version: `any`

Server
- Parse Server version: `any`

### Logs

> type '_Map<String, dynamic>' is not a subtype of type 'List<dynamic>?' in type cast
package:parse_server_sdk/src/objects/parse_base.dart 223:36  ParseBase.get
test/src/objects/parse_object_test.dart 711:39               main.<fn>.<fn>.<fn>
parse_object_test.dart:711

**Comment by Nidal-Bakir** [comment]
I just find it out while writing tests for `ParseObject` class.

I have written all the tests to make sure that will not happen again we just need the fix.

**Comment by mtrezza** [comment]
Thanks @Nidal-Bakir; you've been quite active recently, would you be interested in joining the Parse Flutter SDK review team? You'll get notified when there's a review request for PRs.

**Comment by Nidal-Bakir** [comment]
Thank you for the offer, @mtrezza I would be honored to join the Parse Flutter SDK review team and contribute to the development of the SDK.

**Comment by mtrezza** [comment]
Great, you'll receive an invitation shortly

**Comment by Nidal-Bakir** [comment]
Fix via #860
